src/meta/meta_track.c

```diff
@@ -275,9 +276,17 @@ __wt_meta_track_off(WT_SESSION_IMPL *session, bool need_sync, bool unroll)
 			session, false, WT_TXN_LOG_CKPT_SYNC, NULL));
 		WT_RET(ret);
 	} else {
-		WT_WITH_DHANDLE(session, session->meta_dhandle,
-		    WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_COMMITTED,
-			ret = __wt_checkpoint(session, NULL)));
+		WT_ASSERT(session, F_ISSET(session, WT_SESSION_LOCKED_SCHEMA));
+		ckpt_session = S2C(session)->meta_ckpt_session;
+		/*
+		 * If this operation is part of a running transaction, that
+		 * should be included in the checkpoint.
+		 */
+		ckpt_session->txn.id = session->txn.id;
```

Comment:
It isn't necessary, but I'd feel more comfortable if this ID was cleared after the checkpoint is completed.